Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs

```diff
@@ -374,9 +400,11 @@ public void DirectorySymlinksInOutputDirectory(SealDirectoryKind dirKind)
                     .AssertCacheHit(allPipIds.Except(new[] { consumer.pip.PipId }).ToArray());
             }
 
-            // invalidate producer pip, rerun, expect all cache misses
+            // invalidate producer pip, rerun, expect reads to be cache misses and absent probes to be cache hits 
             InvalidatePip(producerPip);
-            RunSchedulerAndValidateProducedLayout().AssertCacheMiss(allPipIds);
+            RunSchedulerAndValidateProducedLayout()
+                .AssertCacheMiss(readConsumers.Select(t => t.pip.PipId).ToArray())
+                .AssertCacheHit(absentProbeConsumers.Select(t => t.pip.PipId).ToArray());

```

Comment:
>absentProbeConsumers [](start = 32, length = 20)

add a test case where these files become existing as discussed. Thanks for double checking test coverage on this!

Reply:
added (lines 518 and below)